ui/pages/confirmations/components/confirm/network-change-toast/network-change-toast-inner.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { Box } from '../../../../../components/component-library';
+import { Toast } from '../../../../../components/multichain';
+import {
+  getLastInteractedConfirmationInfo,
+  setLastInteractedConfirmationInfo,
+} from '../../../../../store/actions';
+import { getCurrentChainId } from '../../../../../selectors';
+import { NETWORK_TO_NAME_MAP } from '../../../../../../shared/constants/network';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+
+const MILLISECONDS_IN_ONE_MINUTE = 60000;
+const MILLISECONDS_IN_FIVE_SECONDS = 5000;
+
+const NetworkChangeToastInner = ({
+  confirmation,
+}: {
+  confirmation: { id: string };
+}) => {
+  const chainId = useSelector(getCurrentChainId);
```

Comment:
In the interest of multi-chain and future-proofing, should we instead extract the chain ID from the confirmation rather than coupling to the global network?

Reply:
I need to use this value as signature requests do not have chainid in request

Reply:
In that case, could we check the type of confirmation or look for a `chainId` property and only fallback to the selector, so we at least handle transactions with multichain support?